src/coreComponents/physicsSolvers/fluidFlow/IsothermalCompositionalMultiphaseFVMKernelUtilities.hpp

```diff
@@ -1238,11 +1273,24 @@ struct computePotentialGravity
                       Deriv::dC );
 
       // average density and derivatives
-      densMean += 0.5 * density;
-      dDensMean_dPres[i] = 0.5 * dDens_dPres;
+      densMean += density;
+      dDensMean_dPres[i] = dDens_dPres;
       for( localIndex jc = 0; jc < numComp; ++jc )
       {
-        dDensMean_dComp[i][jc] = 0.5 * dProp_dComp[jc];
+        dDensMean_dComp[i][jc] = dProp_dComp[jc];
+      }
+      denom++;
+    }
+    if( denom > 1 )
```

Comment:
Could a phase potentially be absent in both cells? Or is this checked elsewhere?

Reply:
It can be absent in both yes, there is a check below but it would be better to check even outside of the function
Just don't want to add more complexity for this PR